### PR TITLE
MIGRATIONS-1200 - Migrations Repo Cleanup

### PR DESCRIPTION
**Pull request by sumobrian**
### Description
[Describe what this change achieves]
* Category - Code cleanup before release branch is cut
* Why these changes are required? The files removed in this PR are not needed for release. The items renamed we renamed to provide clarity.
* What is the old behavior before changes and new behavior after changes? There are no behavior changes.

Once this PR is approved, a release branch will be cut. On this release branch the additional changes outlined in MIGRATIONS-1200 will addressed.

### Issues Resolved
[List any issues this PR will resolve]
https://opensearch.atlassian.net/browse/MIGRATIONS-1200 

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- Verified end-2-end solution still works.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by mikaylathompson** [comment]
This looks in-progress, but a few notes of other things to be pulled out:
https://github.com/opensearch-project/opensearch-migrations/tree/main/deployment/copilot/traffic-comparator-jupyter
https://github.com/opensearch-project/opensearch-migrations/tree/main/deployment/copilot/traffic-comparator
https://github.com/opensearch-project/opensearch-migrations/tree/main/deployment/docker/traffic-comparator

https://github.com/opensearch-project/opensearch-migrations/blob/main/TrafficCapture/dockerSolution/build.gradle#L12 (various sections)

https://github.com/opensearch-project/opensearch-migrations/blob/main/TrafficCapture/dockerSolution/src/main/docker/docker-compose.yml#L92

**Comment by lewijacn** [comment]
I pictured this as being
```
├── experimental
│   ├── upgrades
│   └── cluster_migration_core
│   └── cluster_traffic_capture
│   └── knowledge_base

```

The one thing that is really throwing me off is that we don't have the top level upgrades directory anymore

**Comment by sumobrian** [comment]
> I pictured this as being
> 
> ```
> ├── experimental
> │   ├── upgrades
> │   └── cluster_migration_core
> │   └── cluster_traffic_capture
> │   └── knowledge_base
> ```
> 
> The one thing that is really throwing me off is that we don't have the top level upgrades directory anymore

This is a good suggestion. I'm going to incorporate these changes.

File: .github/workflows/python-tests.yml [hunk 1]
```diff
@@ -30,8 +30,8 @@ jobs:
           python -m pip install -r requirements.txt coverage pytest-cov
       - name: Run Tests with Coverage
         run: |
-          python -m pytest unit_tests/ --cov=cluster_migration_core --cov-report=xml --cov-branch
+          python -m pytest unit_tests/ --cov=experimental/cluster_migration_core --cov-report=xml --cov-branch
```
Comment (lewijacn) [inline .github/workflows/python-tests.yml:33]: `--cov=cluster_migration_core` as this is using the working-directory above

File: .github/workflows/python-tests.yml [hunk 2]
```diff
@@ -30,8 +30,8 @@ jobs:
...
       - name: Upload Coverage Report
         uses: codecov/codecov-action@v3
         with:
-          files: cluster_migration_core/coverage.xml
\ No newline at end of file
+          files: experimental/cluster_migration_core/coverage.xml
```
Comment (lewijacn) [inline .github/workflows/python-tests.yml:37]: This I am unsure about but might need to be `cluster_migration_core/coverage.xml`

**Comment by lewijacn** [comment]
org/opensearch/migrations/trafficcapture/proxyserver/netty/ExpiringSubstitutableItemPoolTest.java is seeming like a flaky test causing the gradle test to fail, need further RCA

**Review by lewijacn — APPROVED** [review]

**Pull request merged**